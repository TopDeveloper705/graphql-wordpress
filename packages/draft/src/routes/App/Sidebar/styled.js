import styled from 'react-emotion';
import theme from 'styles/theme';
import { headingStyles } from 'styles/utils';

export const Heading = styled.h2`
  ${headingStyles};
  color: ${theme.colors.dark};
  font-family: ${theme.fonts.futura};
  font-size: 25px;
`;

export const Show = styled.div`
  font-size: 15px;
  margin: 0 0 10px 10px;
`;

export const Time = styled.time`
  display: block;
  font-size: 14px;
  font-weight: ${theme.fonts.weight.bold};
`;
