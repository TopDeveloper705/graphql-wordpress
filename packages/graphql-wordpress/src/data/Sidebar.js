import { toGlobalId } from 'graphql-relay';

class Sidebar {
  getID() {
    return toGlobalId(this.constructor.name, this.id);
  }

  static getEndpoint() {
    return process.env.WP_SIDEBARS_ENDPOINT || 'graphql/v1/sidebars';
  }
}

export default Sidebar;
