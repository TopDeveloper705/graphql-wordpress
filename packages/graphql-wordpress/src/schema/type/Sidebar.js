import { GraphQLObjectType, GraphQLList, GraphQLString } from 'graphql';
import WidgetType from 'type/Widget';
import { globalIdField, name } from 'field/identifier';
import description from 'field/description';
import { registerNodeType, NodeInterface } from 'type/relayNode';

const SidebarType = new GraphQLObjectType({
  name: 'Sidebar',
  description: 'A widget area.',
  interfaces: [NodeInterface],
  fields: {
    id: globalIdField(),
    ...name,
    ...description,
    classname: {
      type: GraphQLString,
      description: 'Extra CSS class to assign to the sidebar in the Widgets interface.',
      resolve: sidebar => sidebar.class,
    },
    beforeWidget: {
      type: GraphQLString,
      description:
        "HTML content to prepend to each widget's HTML output when assigned to this sidebar.",
    },
    afterWidget: {
      type: GraphQLString,
      description:
        "HTML content to append to each widget's HTML output when assigned to this sidebar.",
    },
    beforeTitle: {
      type: GraphQLString,
      description: 'HTML content to prepend to the sidebar title when displayed.',
    },
    afterTitle: {
      type: GraphQLString,
      description: 'HTML content to append to the sidebar title when displayed.',
    },
    widgets: {
      type: new GraphQLList(WidgetType),
      description: 'HTML widgets associated with the sidebar.',
    },
  },
});

registerNodeType(SidebarType);

export default SidebarType;
