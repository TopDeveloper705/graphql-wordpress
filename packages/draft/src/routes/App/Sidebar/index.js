import React, { Component } from 'react';
import { graphql } from 'react-apollo';
import gql from 'graphql-tag';
import { Heading, Show, Time } from './styled';

/* eslint-disable react/prop-types */

@graphql(gql`
  query SidebarQuery {
    shows(latest: true) @connection(key: "shows", filter: ["latest"]) {
      edges {
        node {
          id
          artist {
            name
          }
          venue {
            name
          }
          date
        }
      }
    }
  }
`)
export default class Sidebar extends Component {
  state = {};

  render() {
    const { data: { loading, shows } } = this.props;

    if (loading && !shows) {
      return null;
    }

    return (
      <aside>
        <Heading>Upcoming Shows</Heading>
        {shows.edges.map(({ node }) => {
          const d = new Date(node.date);
          const m = d.getMonth() + 1;
          const day = d.getDate();
          return (
            <Show key={node.id}>
              <Time>{`${m < 10 ? `0${m}` : m}/${
                day < 10 ? `0${day}` : day
              }/${d.getFullYear()}`}</Time>
              {node.artist.name}
              <br />
              {node.venue.name}
            </Show>
          );
        })}
      </aside>
    );
  }
}
