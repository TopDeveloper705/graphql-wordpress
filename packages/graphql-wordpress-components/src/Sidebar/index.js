// @flow
import React from 'react';
import { css } from 'emotion';
import styled from 'react-emotion';
import theme from '../theme';
import { uppercaseHeader } from '../';

export const Widget = styled.li`margin: 0 0 ${p => p.theme.padding * 2}px;`;

export const title = css`
  ${uppercaseHeader};
  margin: 0 0 ${theme.padding}px;
`;

export const goToThis = css`
  & .title {
    line-height: 1.6em;
  }

  & a {
    font-weight: ${theme.weightBold};
    text-decoration: none;
  }

  & ol {
    margin: ${theme.padding}px 0;
  }

  & li {
    margin: ${theme.padding / 2}px 0;
    border-top: 3px solid ${theme.colors.detail};
  }
`;

/* eslint-disable react/no-danger */

const transformStyles = (classname, html) =>
  html.replace(/widget-title/g, title).replace(/widget_go_to_this/g, goToThis);

type SidebarProps = {
  sidebar: {
    widgets: Array<Object>,
  },
};

const Sidebar = ({ sidebar }: SidebarProps) => (
  <ul>
    {sidebar.widgets.map(({ id, classname, content: { rendered: widget } }) => (
      <Widget
        key={id}
        dangerouslySetInnerHTML={{
          __html: transformStyles(classname, widget),
        }}
      />
    ))}
  </ul>
);

export default Sidebar;
