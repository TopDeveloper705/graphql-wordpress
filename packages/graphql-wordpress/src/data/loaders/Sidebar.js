import Dataloader from 'dataloader';
import fetchData from 'data/utils';
import Sidebar from 'data/Sidebar';

// Dataloader expects IDs that can be read by the REST API

export default function getSidebarLoaders() {
  const endpoint = Sidebar.getEndpoint();

  const sidebarLoader = new Dataloader(slugs =>
    fetchData(endpoint).then(({ data: { body } }) =>
      slugs.map(id => body.find(item => item.id === id))
    )
  );

  return {
    load: async id => {
      const data = await sidebarLoader.load(id);
      return data ? Object.assign(new Sidebar(), data) : null;
    },
  };
}
