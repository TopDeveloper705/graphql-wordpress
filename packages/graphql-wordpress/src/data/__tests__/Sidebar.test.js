import Sidebar from 'data/Sidebar';

const backupEnv = Object.assign({}, process.env);

afterAll(() => {
  process.env = backupEnv;
});

describe('Test Sidebar data access', () => {
  test('Get endpoint', () => {
    expect(Sidebar.getEndpoint()).toMatchSnapshot();
  });

  test('Get backup endpoint', () => {
    delete process.env.WP_SIDEBARS_ENDPOINT;
    expect(Sidebar.getEndpoint()).toMatchSnapshot();
  });
});
