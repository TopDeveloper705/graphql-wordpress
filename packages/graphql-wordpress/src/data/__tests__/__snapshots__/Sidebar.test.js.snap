// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Sidebar data access Get backup endpoint 1`] = `"graphql/v1/sidebars"`;

exports[`Test Sidebar data access Get endpoint 1`] = `"graphql/v1/sidebars"`;
